enhancements/microshift/optionally-deploy-csi-plugin.md

```diff
@@ -0,0 +1,209 @@
+
+### Workflow Description
+
+**_Prerequisites_**
+
+- Host has LVM installed
+
+**_Installation with CSI Driver and Snapshotting_**
+
+1. User determines there is a requirement for persistent storage and volume snapshotting.
+2. User specifies an ostree blueprint which includes the following sections:
+    1. Packages: microshift, microshift-greenboot, microshift-networking, microshift-selinux, microshift-lvms, 
+       microshift-lvms-snapshotting
+    2. File (Optional): lvmd.yaml
+3. User compiles an ostree commit from the blueprint
+4. User deploys the ostree commit to host
+5. MicroShift host boots
+6. MicroShift starts
+7. MicroShift deploys:
+   1. LVMS CSI manifests
+   2. LVMS Snapshot manifests
+
+**_Installation without CSI Driver and Snapshotting_**
+
+1. User determines there is not a requirement for persistent storage and volume snapshotting.
+2. User specifies an ostree blueprint which includes the following sections:
+    1. Packages: microshift, microshift-greenboot, microshift-networking, microshift-selinux
+3. User compiles an ostree commit from the blueprint
+4. User deploys the ostree commit to host
+5. MicroShift host boots
+6. MicroShift starts
+
+**_Day-1|2 Installation_**
+
+1. User has previous installed MicroShift on a device.
+2. User later determines there is a requirement for persistent storage and volume snapshotting.
+3. User specifies an ostree blueprint which includes the following sections:
+    1. Packages:  microshift-lvms, microshift-lvms-snapshotting
+    2. File (Optional): lvmd.yaml
+4. User compiles an ostree commit from the blueprint
+5. User deploys the ostree commit to host
+6. MicroShift host reboots
+7. MicroShift starts
+8. MicroShift deploys:
+   1. LVMS CSI manifests
+   2. LVMS Snapshot manifests
+
+### API Extensions
+
+- None
+
+### Topology Considerations
+
+#### Single-node Deployments or MicroShift
+
+The changes proposed here only affect MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+- **Rebase Changes:** 
+  - LVMS and the snapshotting manifests will be moved from `microshift/assets/components/lvms` and 
+  `microshift/assets/components/csi-snapshot-controller` to `microshift/assets/optional/lvms` and 
+  `microshift/assets/optional/csi-snapshot-controller`. These paths will be updated in the 
+    `microshift/scripts/auto-rebase/lvms_assets.yaml` to reflect the new paths.
+  - LVMS image digests will be moved from `microshift/assets/release/release-$ARCH.json` to 
+    `microshift/assets/optional/lvms/release-$ARCH.json`.
+  - CSI Snapshotter image digests will be moved from `microshift/assets/release/release-$ARCH.json` to 
+      `microshift/assets/optional/csi-snapshot-controller/release-$ARCH.json`.
+  - During rebasing, the image digests for LVMS and CSI Snapshotter will be parsed from the LVMS bundle (like they 
+    are now) and the digests will be written to their respective `release-$ARCH.json` files.
+
+- **Mainline Code Changes:**
```

Comment:
I really think this needs to be revisited because this logic still has to exist somewhere.

Reply:
I've added a section called RPM-Spec about this. LMKWYT